Performance optimizations for large monorepo

ejlangev opened the pull request:
I've been trying to run ember-cli-update on a large monorepo and ran into some problems. First I addressed the functional issue I found with git command buffers with PR https://github.com/kellyselden/git-diff-apply/pull/277 which was a result of run `git ls-files` on a repo with more than 15k files in it.  After merging that I found that while ember-cli-update was working it was extremely slow which I figured was because it ran a large number of git commands.

This pull request addresses two things:

1) Before these changes it needed to run `git add <file-path>` on more than 15k files individually which took a very long time.  Batching the files within a certain command size (started with 4k characters each) sped it up considerably by reducing the number of git commands to run to under 500.

2) `git status --porcelain` could have a very large output size because many files could be in the git staging area after adding. I fixed this by using the spawn based command runner that I added in the previous pull request.

I hope this is helpful, trying to get to the point where I can easily run `ember-cli-update` on my mid-sized ember project that is embedded in a large monorepo without issues and it seems to work pretty well with these changes.

kellyselden commented on src/git-remove-all.js:
```diff
@@ -16,20 +15,41 @@ async function lsFiles(options) {
   return files;
 }
 
+function chunkFilePaths(files, maxChunkSize = 4096) {
```
Perhaps we should get a unit test or two on this to prove that it is chunking as expected. I would hate to tweak this in the future and accidentally break it.

ejlangev replied:
👍good idea, I'll make one and then update

kellyselden commented:
#280 should prevent this code from regressing.

ejlangev commented:
Okay @kellyselden I added some unit tests.  I'm glad you suggested it because I found an edge case that was not handled properly and fixed it. Good idea on the performance regression test as well 👍.

kellyselden commented:
Thanks again!
